### ISSUE TITLE
Getting reservations with the proper scope gives me 403

### ISSUE THREAD

**Issue by rockymontana** [apaleo/api#423]
Hello Apaleo.

I'm currently building an integration for our application.
However, when I try to fetch the reservations I get forbidden.

This is the scopes I've got access to (reservations marked as bold):
"scope" => "account.manage accounting.read availability.manage availability.read companies.manage companies.read distribution:reservations.manage distribution:subscriptions.manage folios.manage folios.payment-with-charges folios.read identity:account-users.manage identity:account-users.read integration:ui-integrations.manage invoices.manage invoices.read logs.read maintenances.manage maintenances.read offer-index.read offers.read openid operations.change-room-state operations.trigger-night-audit profile profile:manage profile:read rateplans.read-corporate rateplans.read-negotiated rates.manage rates.read reports.read **reservations.force-manage reservations.manage reservations.read** setup.manage setup.read"

A matter of fact is that whatever endpoint I'm trying to reach I get a 403.
I'm not sure what I'm doing wrong here.

Can you point me in the right direction?
Thanks

**Comment by WinniX** [comment]
Hello @rockymontana 

Could you share the `TrackingId` with us? So we could faster identify your request.

Eugene / apaleo Team

**Comment by rockymontana** [comment]
I don't get a trackingID that I can see (unless it's in the header which I haven't checked). I only get a 403 and no content.

**Comment by WinniX** [comment]
Hello @rockymontana 

Let me share some guides with you, please make sure your setup is correct:
- https://apaleo.dev/guides/start/oauth-connection/auth-code-grant => for connect flow (which is the recommended if you are making an integration that is planned to be distributed to multiple hotels as a public app)
- https://apaleo.dev/guides/start/oauth-connection/client-credentials-grant => if you are creating an internal/private app for a specific hotel which is not expected to be used as a public app

Hope this helps. Let me know if you still have any questions.

--
Eugene / apaleo Team

**Comment by rockymontana** [comment]
Hello Eugene.
I've been following the second article and I do get a token back.
In my opening post I showed what scopes I get back from you when I get my token.
I can even set the specific scope I want a token for (reservations in this case) and I get a response back with a token and that scope. So the authorization works. Otherwise I would have gotten 401 Unauthorized (if I read your Swagger documentation correctly).
I get 403 Forbidden back though, which seems to me like I don't have the proper permissions inside the application to do specific tasks, but I'm still logged in.

![image](https://user-images.githubusercontent.com/330839/97538173-452e2780-19c0-11eb-9366-afbca83d8c11.png)

![image](https://user-images.githubusercontent.com/330839/97538355-86263c00-19c0-11eb-88ae-4b5bc32631f6.png)
(this regardless if I use /v1 or /v0-nsfw endpoints)

**Comment by rockymontana** [comment]
Found a trackingId in the header:
6fd22ae7-8a04-4c2b-9288-e50352efd10a

**Comment by rockymontana** [comment]
Any ideas?

**Comment by randyrod** [comment]
Hi @rockymontana,

You mentioned that you are following the second guide that @WinniX shared, which is the flow for the simple clients. However, you are using a connect client. So even though you are getting a token, in reality it is an invalid token.

In order to get a valid token, you must follow the steps described in this guide https://apaleo.dev/guides/start/oauth-connection/auth-code-grant

Hope this information is helpful.

**Comment by rockymontana** [comment]
How come that when I follow the guide for a simple client, I retrieve connect client credentials?
Did I do something wrong when I created the client secret stuff? Can I just create another pair or client id and secret and use that to get a simple client? I didn't see any options to create either.

I'm build a server to server solution, so having to do the auth-code-grant solution is a bit tedious since I don't have a browser or a frontend and it would require a lot more for our integration.
So I ask this instead:
What can I do to just get a valid token back (obviously I've already followed your simple token guide)?
